service/src/main/java/org/mskcc/cmo/metadb/service/impl/SampleServiceImpl.java

```diff
@@ -171,4 +166,15 @@ public Boolean sampleHasMetadataUpdates(SampleMetadata existingSampleMetadata,
         }
         return Boolean.FALSE;
     }
+
+    @Override
+    public PublishedMetadbSample getPublishedMetadbSample(UUID metadbSampleId) throws Exception {
+        MetadbSample sample = getMetadbSample(metadbSampleId);
+        PublishedMetadbSample publishedSample = new PublishedMetadbSample(sample);
```

Comment:
here we are passing the `MetadbSample` along to the `PublishedMetadbSample` constructor and then making a subsequent call to `publishedSample.setMetaDbSampleId()` 

since `MetadbSample` is being passed to the constructor anyway you can set the metadb sample id within the constructor itself instead of making this extra call here

Reply:
same comment for setting up the patient stuff - all of this information is already available to the constructor

Reply:
When calling [getMetadbSample](https://github.com/mskcc/cmo-metadb/pull/468/files#diff-14fbd1dc6e948619c6d34efb8f41005965101954a8895440857979d916eb6b57R172) does it not also get and set the `MetadbPatient`?

I see metadb patient id [isn't getting set anymore](https://github.com/mskcc/cmo-metadb/pull/468/files#diff-14fbd1dc6e948619c6d34efb8f41005965101954a8895440857979d916eb6b57L99) which i agree with but i did make a comment somewhere suggesting that we still set up the `MetadbPatient` field for the `MetadbSample` anywhere that we are constructing an instance of `MetadbSample`

Reply:
once that is implemented then the patient service wouldnt need to be called in this method and we can move forward with the constructor containing all of the initialization/set up logic for creating a `PublishedMetadbSample`

Reply:
[reference](https://github.com/mskcc/cmo-metadb/pull/468/files#diff-14fbd1dc6e948619c6d34efb8f41005965101954a8895440857979d916eb6b57L130)